server/require.js

```diff
@@ -42,7 +42,8 @@ export function getPagePath (page, {distDir}) {
     throw pageNotFoundError(page)
   }
 
-  return join(serverBuildPath, pagesManifest[page])
+  // Replace any separators with the current platform separator
+  return join(serverBuildPath, pagesManifest[page].replace(/(\/|\\)/g, sep))
```

Comment:
Hmm. I was gonna say that we could do the fixing when loading file, but we load the json file for each request too. Could we require the json file and fix paths as some init step that's only run once?

Then again, [forward slash should work almost always on windows](https://superuser.com/questions/176388/why-does-windows-use-backslashes-for-paths-and-unix-forward-slashes/176395#176395). (["Forward-slash paths can be used in Windows as long as they're not extended-length paths and don't contain any non-ascii characters."](https://github.com/sindresorhus/slash)). So you could potentially skip it completely.

Reply:
You can use this instead I think `.replace(/\\/g, '/')`

Reply:
Actually it's better to replace here than to loop over the whole manifest and fix the paths, as this would be faster in a serverless environment.

Reply:
> You can use this instead I think `.replace(/\\/g, '/')`

Ok so is that for backwards compability then? Since the changes to `pages-mainfest-plugin.js` should ensure that all paths are forwardslashes already?

Reply:
Right I meant without the changes in the manifest plugin, basically the issue here is that we can't `join()` windows and unix paths. `require` accepts unix paths for windows, but we can't take advantage of that because of the join.

Reply:
> Right I meant without the changes in the manifest plugin, basically the issue here is that we can't `join()` windows and unix paths. `require` accepts unix paths for windows, but we can't take advantage of that because of the join.

What do you mean we can't take advantage of `require('unix/path')` because we are using join? 
On windows `join('windows\\path', 'posix/path')` => `windows\\path\\posix\\path`. If the changes in `pages-manifest-plugin.js` are implemented we could actually skip slash replacement entirely here. It wouldn't be backwards compatible with pages-manifest.json files still containing backslashes however.

If we don't want the changes in pages-manifest-plugin for whatever reason, then your suggestion is correct yes. Why would we not want pages-manifes-plugin to handle the replacement on buildtime rather than dealing with it on runtime?

Reply:
Ohhhhhhhhh now I understand, so let's remove this then, only changing the manifest is enough, people have to rebuild their app anyway 👍